### ISSUE TITLE
Why does XGBoost predict the same value for all

### ISSUE THREAD

**Issue by huanzuo66** [dmlc/xgboost#3381]
I don't know why running the XGB prediction model in R to predict binary result that comes out the model always produces the exact same result. I couldn't find a good answer for it. If anybody can help me?

##Selecting 70% of data as sample from total 'n' rows of the newsparse as training data
sample <- sample.int(n = nrow(newsparse), size = floor(.7*nrow(newsparse)), replace = F)
mytrain <- newsparse[sample, ]
mytest  <- newsparse[-sample, ]

##Creating the matrix for training the model
ctrain <- xgb.DMatrix(Matrix(data.matrix(mytrain[,!colnames(mytrain) %in% c('RootCause_Bin')])), label = as.numeric(mytrain$RootCause_Bin))

> table(mytrain$RootCause_Bin)

   0    1 
1652 3328 

> table(mytest$RootCause_Bin)

   0    1 
 719 1416 

#predict
xgbmodel <- xgboost(data = ctrain, max.depth = 15, eta = 0.3, nround = 100, objective = "binary:logistic",booster="gbtree", subsample=0.8, scale_pos_weight=0.5)

#get prediction classification
mytest_label <- ifelse (xgbmodel.predict < 0.5,"0","1")

> table(mytest_label)
mytest_label
   1 
2135
